snuba/clusters/storage_sets.py

```diff
@@ -29,3 +30,25 @@ class StorageSetKey(Enum):
 
 # Storage sets enabled only when development features are enabled.
 DEV_STORAGE_SETS: FrozenSet[StorageSetKey] = frozenset({StorageSetKey.METRICS})
+
+# Storage sets in a group share the same query and distributed nodes but
+# do not have the same local node cluster configuration.
+# Joins can be performed across storage sets in the same group.
+STORAGE_SET_GROUPS: FrozenSet[Tuple[StorageSetKey, ...]] = frozenset(
```

Comment:
Do you envision any use case for this abstraction other than adding constraints to joins? 
I think we should be clear on the semantics of what this is for so that it does not become a catch all abstraction used for any sort of grouping we want to build. 
(That will be bad because we will end up in weird configuration if we start having storage set groups used for different reasons at the same time).

If this is just about joins, I would call them JOINABLE_STORAGE_SETS and maybe just move it together with the join modules instead of making it something part of the storage set infrastructure.

Reply:
Also why a tuple and not a Set? Are the storage sets in a group supposed to be in order ? If yes why?

Reply:
Yeah, I can't think of any other use case except queries that span multiple tables (which at this point is just joins)

Reply:
Order is definitely not important here, I probably did it this way since set is a not hashable type so you can't have a set of sets. Though looking at this again, I could convert the tuple into a frozen set to get around this.

Reply:
Renamed to JOINABLE_STORAGE_SETS and made it a frozenset. I think we should leave it here rather than nesting under `snuba.query.joins` module.